lang/csharp/src/apache/main/IO/BinaryDecoder.cs

```diff
@@ -298,6 +309,10 @@ private void Skip(long p)
         }
 
         /// <inheritdoc />
-        public void Dispose() => stream?.Dispose();
+        public void Dispose()
+        {
+            if(!ownStream)
```

Comment:
This condition IMO should be `if (ownStream) { stream?.Dispose(); }`

Reply:
Since the ownStream = false in the default constructor, the stream would be disposed using the default constructor, which woulr make it a breaking change.